File: baselibrary/src/main/java/com/shiye/baselibrary/net/loadingView/LoadingDialog.java
package com.shiye.baselibrary.net.loadingView;

import android.app.AlertDialog;
import android.content.Context;
import android.os.Bundle;
import android.view.View;

import com.shiye.baselibrary.R;

/**
 * Created by issuser on 2018/7/19.
 */
public class LoadingDialog extends AlertDialog {
    private View mLoadingView;

    public LoadingDialog(Context context) {
        this(context, -1);
    }

    public LoadingDialog(Context context, int themeResId) {
        super(context, themeResId);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (mLoadingView != null){
            setContentView(mLoadingView);
        }else {
            setContentView(R.layout.loading_dialog);
        }
    }
    public void setLoadingView(View loadingView){
        mLoadingView = loadingView;
    }
}

File: baselibrary/src/main/java/com/shiye/baselibrary/net/loadingView/LoadingDialog.kt
package com.shiye.baselibrary.net.loadingView

import android.app.AlertDialog
import android.content.Context
import android.os.Bundle
import android.view.View

import com.shiye.baselibrary.R

/**
 * Created by issuser on 2018/7/19.
 */
class LoadingDialog @JvmOverloads constructor(context: Context, themeResId: Int = -1) : AlertDialog(context, themeResId) {
    private var mLoadingView: View? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mLoadingView?.let { setContentView(it) }
        setContentView(R.layout.loading_dialog)
    }

    fun setLoadingView(loadingView: View?) {
        mLoadingView = loadingView
    }
}
